doc/release/upcoming_changes/23240.compatibility.rst

```diff
@@ -0,0 +1,6 @@
+Array-likes that define ``__array_ufunc__`` can now override ufuncs if used as the ``where`` keyword argument
+---------------------------------------------------------------------------------------------------
```

Comment:
Would love to shorten, but seems fine.  I think some parsers might not like the underline to have a different length (but I am not sure).

Reply:
Ok, I shortened it to just "Array-likes that define ``__array_ufunc__`` can now override ufuncs if used as ``where``" since I think it's pretty obvious that we're talking about the keyword argument.